pkg/config/config.go

```diff
@@ -295,6 +313,8 @@ func NewConfigFromMap(data map[string]string) (*Config, error) {
 		cm.AsBool(EnableMeshPodAddressabilityKey, &nc.EnableMeshPodAddressability),
 		cm.AsString(DefaultExternalSchemeKey, &nc.DefaultExternalScheme),
 		cm.AsBool(InternalEncryptionKey, &nc.InternalEncryption),
+		cm.AsString(InternalDataplaneTrustKey, &nc.InternalDataplaneTrust),
```

Comment:
Should we add the validation under L#323?

Reply:
I agree that we should. We would not want a 'disabled' default for cases where someone makes a type and writes 'enbled'. we also cant have 'enabled' anything with tls as default - so the bottom line, we can't have a default which means we need to verify that one of the legal values is used.